### DISCUSSION TITLE
alternate namespace / secand cluster?

### DISCUSSION THREAD

**Discussion by caffeinism** [rook/rook#11381] [Q&A] [answered]
In rook's recent documentation there is a manual for deploying the second cluster. (https://rook.io/docs/rook/v1.10/Storage-Configuration/Advanced/ceph-configuration/#using-alternate-namespaces) I have a few questions about this.

1. A script that replaces multiple words is introduced. But some comments are not present in the file.
The following comments are completely deprecated. Is this intentional? Or will it be added later or is it deprecated?
```
# serviceaccount:namespace:operator
# serviceaccount:namespace:cluster
# driver:namespace:cluster
```

2. Curious about multiple rolebindings
There are several rolebindings in common.yaml. But common-second-cluster.yaml seems to be missing something. Is this good enough? (Is it possible to have only one csi-related role binding across multiple clusters? Is it sufficient to have an operator for each namespace? etc.)

Sorry for my rudimentary question. thank you

**Comment by caffeinism** [comment, marked as answer]
After thinking about it for a bit, I realized that I didn't have to worry about this. I came to the idea that I only need to manage one csi across the cluster anyway, and only the cluster can be managed per namespace.

**Reply by travisn** [reply]
Correct, it's recommended to have a single operator and csi driver, even if there are ceph clusters in multiple namespaces.

**Comment by caffeinism** [comment]
After testing it, other questions keep popping up. If I use one operator (+csi related plugins) when deploying multiple rook-ceph clusters, I think it is correct to separate the namespace of the operator and the namespace of the cluster. (e.g. operator deploys to rook-ceph-operator, cluster1 deploys to rook-ceph-1, cluster2 deploys to rook-ceph-2)

This is actually feasible as shown in the above document, but it would be better if common.yaml separates operator-related definitions from cluster-related definitions. (Example: common-operator.yaml, commont-cluster.yaml)

**Reply by travisn** [reply]
Since the most common case is to have a single ceph cluster, for simplicity the assumption is made in the examples that the operator and cluster are in the same namespace. Then any additional ceph instances would be added to another namespace. But you could certainly choose to run the operator in a separate namespace from either cluster.

Note that the helm chart also is expected to take care of all the RBAC for you in different namespaces if you choose to install multiple ceph clusters. We haven't put effort into factoring out the common.yaml further since the helm chart can be used for this purpose.

**Reply by caffeinism** [reply]
I looked up the documentation and realized that the Helm charts were properly divided, so I changed all the distribution specs with it. It seems like the right choice.
